tests/lib/enableRendererCoverage.js

```diff
@@ -0,0 +1,273 @@
+ * This function is sent to renderer processes (as a string) and run there.
+ * Once the coverage server `gpii.testem.coverage.express` returns a response it notifies the
+ * main process so that the tests would continue.
+ *
+ * NOTE: The BrowserWindow has to have "node integration" enabled in order to collect coverage data.
+ * A work around would be partly loading code with window init - webContents preload script.
+ */
+gpii.tests.app.sendRendererCoverage = function () {
+    require("../../../../node_modules/gpii-testem/src/js/client/coverageSender");
+
+    function notifyCoverageReportSuccess() {
+        require("electron").ipcRenderer.send("coverageReportSuccess");
+    }
+
+    var options = {
+        coveragePort: 7003,
+        exposeCallback: true
+    };
+    window.gpii.testem.coverage.sender(options);
+    window.gpii.testem.coverage.afterTestsCallback(null, null, notifyCoverageReportSuccess);
+};
+
+
+/**
+ * Send the coverage collecting code to the specified `BrowserWindow`. Note that the code is sent
+ * as a string. Refer to `gpii.tests.app.sendRendererCoverage` for details.
+ * @param {Component} dialog - The `BrowserWindow` instance
+ */
+gpii.tests.app.instrumentedDialog.requestDialogCoverage = function (dialog) {
+    var sendCoverageCommand = fluid.stringTemplate("(%function)()", { function: gpii.tests.app.sendRendererCoverage.toString() });
+    gpii.test.executeJavaScript(dialog, sendCoverageCommand);
+};
+
+/**
+ * Request coverage data from all BrowserWindows that currently exists (hopefully all dialogs created throughout the test
+ * sequence). It is resolved once all windows have notified that their coverage had been sent successfully which will allow
+ * next test sequence to be executed. Tests notify their success on reporting coverage over the IPC using
+ * the predefined socket - "coverageReportSuccess".
+ *
+ * NOTE that the coverage data is collected by executing a command sent from the main process - `gpii.tests.app.sendRendererCoverage`.
+ * For more details refer to gpii.test.executeJavaScript.
+ *
+ * @return {Promise} A promise that is resolved once all the BrowserWindows have repoted their coverage back to the
+ * coverage server.
+ */
+gpii.tests.app.instrumentedDialog.requestCoverage = function () {
+    var promise = fluid.promise();
+
+    var activeDialogs = BrowserWindow.getAllWindows();
+    var awaitingDialogReports = activeDialogs.length;
+
+    fluid.log("Collect coverage from active dialogs: ", activeDialogs.length);
+
+    fluid.each(activeDialogs, function (dialog) {
+        var collectDialogCoverage = gpii.tests.app.instrumentedDialog.requestDialogCoverage.bind(null, dialog);
+
+        /*
+         * It might be the case that a test sequence has finished execution before all BrowserWindows have been fully
+         * initialized. In that case we should wait for the BrowserWindows to be ready.
+         */
+        if (!dialog.domReady) {
```

Comment:
Whenever there is an if-else statement, I think it makes it harder to ready when the if-condition contains a negation. I suggest swapping the if and else blocks and making the condition simply `dialog.domReady`.